aten/src/ATen/native/Normalization.cpp

```diff
@@ -240,7 +240,8 @@ std::tuple<Tensor, Tensor, Tensor> batch_norm_backward_cpu_template(
     grad_weight = at::empty_like(weight, at::MemoryFormat::Contiguous);
   }
   if (grad_input_mask[2]) {
-    grad_bias = at::empty_like(weight, at::MemoryFormat::Contiguous);
+    auto options = input.options();
+    grad_bias = at::empty({input.size(1)}, input.options());
```

Comment:
Why do you need to do that?
Also this might not be contiguous anymore?

Reply:
`weight` could be undefined. We run into issue here with BN with only bias. e.g.

https://github.com/pytorch/pytorch/pull/57321/files#diff-4a85700059a0d89291a89a82d2f1ce111293a80c5bc7009b0479fb51b7e5d548R119-R121

> Also this might not be contiguous anymore?
We are creating fresh output 1d buffer, I don't quite get which is non-contiguous in this context.

Reply:
cc @zou3519 could this be breaking some existing batched Tensor code?
How do we deal with this case where weight might be undefined, should we just check and call different functions?

Reply:
There is no problem for existing BatchedTensor code because native_batch_norm is a primitive w.r.t. autograd (it has a backwards formula) BatchedTensor never sees this code.